Feature/dx 119 (#5)
SwedbankPay/developer.swedbankpay.com: payments/index.md

---
title: Swedbank Pay Payments
sidebar:
  navigation:
  - title: Payments
    items:
    - url: /payments/
      title: Introduction
    - url: /payments/credit-account
      title: Credit Account Payments
    - url: /payments/credit-card
      title: Credit Card Payments
    - url: /payments/invoice
      title: Invoice Payments
    - url: /payments/direct-debit
      title: Direct Debit Payments
    - url: /payments/mobile-pay
      title: Mobile Pay Payments
    - url: /payments/swish
      title: Swish Payments
    - url: /payments/vipps
      title: Vipps Payments
---

{% include alert.html type="warning"
                      icon="warning"
                      header="Site under development"
                      body="The Developer Portal is under construction and should not be used to integrate against Swedbank Pay's APIs yet." %}

> # The Fundamentals

Reviewer: No first-level headings, please. This can be removed.

> >eCommerce APIs share a common foundation with a similar payment process for all payment methods, which reduce complexity and enable a more straightforward integration.

Reviewer: ```suggestion
>All Payments APIs in the Swedbank Pay API Platform share a common foundation with a similar payment process for all payment instruments, reducing complexity and enabling a more straightforward integration.
```

+ Authorize funds.  
  An authorization transaction reserves the funds necessary funds . It is possible to abort a payment before the end user has fulfilled the payment process.  
And either:  
+ Capture funds.  
  Before delivering the merchandise you need to create a capture transaction to ensure that the money is charged from the consumer credit card or properly billed by invoice.  
Or:  
+ Cancel the authorized amount.  
 Funds that are authorized but not yet captured can be released back to the consumer. This is done by creating a cancel transaction. 
+ Reverse captured funds.  
 In some cases you may need to make a reversal of captured funds. This is achieved by creating a reversal transaction.

*One-phase payments* ([Direct Debit](/payments/direct-debit), [Swish](/payments/swish), [Card](/payments/credit-card))

If you use a one-phase method (like Direct Bank Debit or [Swish](/payments/swish)) a Sales transaction will be created and the consumer charged right away. 

* _One-phase card payments is enabled with autocapture._

>Add image

> ### <a name="creating-a-payment-object">Creating a payment object</a>.

Reviewer: We shouldn't need to use `<a name>` as Jekyll will create `id` attributes on all headers for us.

```suggestion
### Creating a payment object
```

The `payment` is the container object that holds all transactions that will be created during the payment process. When PayEx receives the payment request body (in JSON format), a payment  is created and you will be given a unique paymentID in return. The response also include  (in a true RESTful way) the URIs to possible further actions, given the state of the payment. It is possible to abort a payment before the end user has fulfilled the payment process.
Settle funds directly with a Sales transaction.  
```

> Pick and choose between the payment methods best suited for your business. Take advantage of our easy-to-use PCI compliant platforms Redirect and PayEx Hosted View - or use PayEx Direct API to integrate directly. Our payment methods and their platform availability are listed in the table below.

Reviewer: This looks more like prose than code to me, so wrapping it in code fences does not feel right. A quote `>` might be more suitable?

```

{:.table .table-striped}
|  | Payment method | Hosted View | Redirect | Direct API | Region |
| CardIcon | [Credit card](payment/credit-card/) | ✔︎︎︎︎︎ | ✔︎︎︎︎︎ | ✔︎︎︎︎︎ | EarthIcon |
| InvoiceIcon | [PayEx Invoice](payment/invoice/) | ✔︎︎︎︎︎ | ✔︎︎︎︎︎ | ✔︎︎︎︎︎ | NordicCountries |
| EnvelopeIcon | [PayEx Web Invoice](payment/credit-account) | ✔︎︎︎︎︎ | ✔︎︎︎︎︎ | ✔︎ ︎| SwedenNorway|
| KeypadIcon | [Direct Debit](payment/direct-debit)| X | ✔︎︎︎︎︎ | X | BalticNordic |
| Vipps | [Vipps](payment/vipps) | ✔︎︎︎︎︎ | ✔︎︎︎︎︎ | X | Norway |
| Swish | [Swish](payments/swish) | ✔︎︎︎︎︎ | ✔︎︎︎︎︎ | ✔︎︎︎︎︎  ︎| Sweden |
| MobilePay | [Mobile Pay](payments/mobile-pay) | X | ✔︎︎︎︎︎ | X | Denmark |

>Add image

## PayEx Payment Instruments Platforms
PayEx Hosted view and Redirect offer easy-to-use PCI compliant platforms, available from both web and mobile browsers. Either let your customers access the platform directly, embedded on your site, or by redirection to a separate hosted payment page.

> ### Hosted View implementation

Reviewer: I've realized that we need to create one section per integration variant:

1. Redirect
2. Seamless View (aka Hosted View)
3. Tailored (for Checkout)

Each section needs to describe its integration variant fully, from A to Z.

Author: Okay, should we do that now or later?

Reviewer: Perhaps we can do it in a separate task and PR. It might involve a lot of work.

```
TODO: Insert pickture from demoshop showing the following configuration:  
PayEx Payment Pages  
Redirect  
Norway  
English
```

#### Getting started
To start using the Redirect platform, you need to implement the appropriate operation returned from the initial API Request, containing the URL that the customer's browser needs to be redirected to.

The combination of properties should be similar to all payment methods. In the example below, the href attribute refers to the redirect URL and the rel description redirect-sale indicate that the redirect scenario will generate a one-phased sales transaction. A two-phase credit card payment would during the same scenario generate an authorization transaction, and the  rel description would in that case be `"redirect-authorization"`.

```JS
{
   "operations": [
        {
           "method": "GET",
           "href": "https://ecom.externalintegration.payex.com/swish/payments/sales
                     /993b479653da83671c074316c7455da05fced9d634431edbb64f3c5f80a863f0",
           "rel": "redirect-sale"
        }
    ]
}
```
See the technical overview of each payment method and the technical reference for more specific information.

> ## <a name="the-payment-process">The Payment Process</a>

Reviewer: ```suggestion
## The Payment Process
```

The core payment process of the eCommerce APIs are the same across all payment instruments.

> 1. ### Creating a payment

Reviewer: Headers in lists are a weird construction. Let's just use headers for the top-level items and keep the lists underneath each header.

```suggestion
### Creating a payment
```
